cloudant-client/src/main/java/com/cloudant/client/api/Database.java

```diff
@@ -177,6 +178,7 @@ public void setPermissions(String userNameorApikey, EnumSet<Permissions> permiss
             String ok = getAsString(response, "ok");
             if (!ok.equalsIgnoreCase("true")) {
                 //raise exception
+                throw new CouchDbException("Error setting permissions.");
```

Comment:
I considered it, but was trying to keep the fix small. To make that change requires a bit more re-working because the response stream has already been consumed and potentially hidden any `error` or `reason` JSON from us. Maybe a future fix?